keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -333,36 +336,170 @@ or later. The dual-stack feature requires no changes to this API.
 The versions of CNI plugin binaries that must be used for proper dual-stack functionality (and IPv6 functionality in general) depend upon the version of the container runtime that is used in the cluster nodes (see [CNI issue #531](https://github.com/containernetworking/cni/issues/531) and [CNI plugins PR #113](https://github.com/containernetworking/plugins/pull/113)):
 
 ### Services
+Services depend on `--services-cluster-ip-range` for `ClusterIP` assignment. A dualstack cluster can have this flag configured as `--services-cluster-ip-rang=<cidr>,<cidr>` or `--services-cluster-ip-rang=<cidr>`. The following rules apply:
+
+1. The maximum allowed number of cidrs in this flag is two. Attempting to use more than two cidrs will generate failures during startup. 
+2. When configured with two cidrs then the values are expected to have different ip families. e.g `--services-cluster-ip-range=<ipv4-cidr>,<ipv6-cidr>` or `--services-cluster-ip-range=<ipv6-cidr>,<ipv4-cidr>`. Attempting to configure a cluster with two similar cidrs will generate failures during startup.
+3. A dualstack cluster can have a single cidr in this flag. That means the cluster will be to egress traffic on both ip families, but will *not* be ingress traffic on both ip families.
+4. When upgrading a cluster to dualstack, the first element of the `--services-cluster-ip-range` flag MUST NOT change IP families.  As an example, an IPv4 cluster can safely be upgraded to use `--services-cluster-ip-range=<ipv4-cidr>,<ipv6-cidr>`, but can not safely be changed to `--services-cluster-ip-range=<ipv6-cidr>,<ipv4-cidr>`.
+<<<<<<< HEAD
```

Comment:
git rebase leftover ??

Reply:
yeah. things got messed up and had to fix manually. will remove. thanks